pkg/model/api/v1/ephemeraldashboard.go

```diff
@@ -0,0 +1,147 @@
+// Copyright 2024 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+
+	modelAPI "github.com/perses/perses/pkg/model/api"
+	"github.com/prometheus/common/model"
+)
+
+type EphemeralDashboardSpecBase struct {
+	TTL model.Duration `json:"ttl" yaml:"ttl"`
+}
+
+func (edsb *EphemeralDashboardSpecBase) UnmarshalJSON(data []byte) error {
+	var tmp EphemeralDashboardSpecBase
+	type plain EphemeralDashboardSpecBase
+	if err := json.Unmarshal(data, (*plain)(&tmp)); err != nil {
+		return err
+	}
+	*edsb = tmp
+	return nil
+}
+
+func (edsb *EphemeralDashboardSpecBase) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	var tmp EphemeralDashboardSpecBase
+	type plain EphemeralDashboardSpecBase
+	if err := unmarshal((*plain)(&tmp)); err != nil {
+		return err
+	}
+	*edsb = tmp
+	return nil
+}
+
+type EphemeralDashboardSpec struct {
+	EphemeralDashboardSpecBase
```

Comment:
and what about the Marshal, it is creating the right structure ?? I'll never understand all these oddities...

Reply:
I see in the following tests that the answer of my question is yes....